src/applications/appeals/995/containers/ConfirmationPage.jsx

```diff
@@ -12,111 +12,118 @@ import { FORMAT_READABLE } from '../constants';
 import { getSelected, getIssueName } from '../utils/helpers';
 import GetFormHelp from '../content/GetFormHelp';
 
-export class ConfirmationPage extends React.Component {
-  componentDidMount() {
-    scrollTo('topScrollElement');
-    setTimeout(() => {
-      focusElement('va-alert h2');
-    });
-  }
+export const ConfirmationPage = () => {
+  const alertRef = useRef(null);
 
-  render() {
-    const { name = {}, form } = this.props;
-    const { submission, data } = form;
-    const issues = getSelected(data || []).map((issue, index) => (
-      <li key={index} className="vads-u-margin-bottom--0">
-        {getIssueName(issue)}
-      </li>
-    ));
-    const fullName = `${name.first} ${name.middle || ''} ${name.last}`;
-    const submitDate = moment(submission?.timestamp);
-    resetStoredSubTask();
+  const form = useSelector(state => state.form || {});
+  const name = useSelector(state => selectProfile(state)?.userFullName || {});
 
-    return (
-      <div>
-        <div className="print-only">
-          <img
-            src="https://www.va.gov/img/design/logo/logo-black-and-white.png"
-            alt="VA logo"
-            width="300"
-          />
-          <h2>Application for Supplemental Claim</h2>
-        </div>
-        <va-alert status="success">
-          <h2 slot="headline">Thank you for filing a Supplemental Claim</h2>
-          <p>
-            When we’ve completed our review, we’ll mail you a decision packet
-            with the details of our decision.
-          </p>
-        </va-alert>
-        <div className="inset">
-          <h3 className="vads-u-margin-top--0">
-            Your information for this claim
-          </h3>
-          <h4>Your name</h4>
-          {fullName ? (
-            <div>
-              {name.first} {name.middle} {name.last}
-              {name.suffix ? `, ${name.suffix}` : null}
-            </div>
-          ) : null}
+  useEffect(
+    () => {
+      if (alertRef?.current) {
+        scrollTo('topScrollElement');
+        setTimeout(() => {
+          // delay focus for Safari
+          focusElement('h2', {}, alertRef.current);
+        }, 250);
```

Comment:
We had a similar issue with one of our pages on Profile and I used the `waitForRenderThenFocus` with a low interval (i think 50) to make sure that focus was correct on Safari

Do you think there is a benefit in one over the other?

Reply:
I didn't think to use that function LOL.. I think `waitForRenderThenFocus` might be better